ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/GeneralizeMaskingProvider.java

```diff
@@ -54,86 +42,14 @@ public GeneralizeMaskingProvider() {
-      ObjectMapper mapper = ObjectMapperFactory.getObjectMapper();
-      JsonNode ruleSetNode = mapper.readTree(ruleSetStr);
-      if (ruleSetNode != null && !ruleSetNode.isMissingNode()) {
-
-        for (JsonNode ruleNode : ruleSetNode) {
-          /*
-           * Validate the rule has a targetValue node, and also has either sourceValueIn or
-           * valueSourceNotIn, but not both. Currently, must specify either sourceValueIn or
-           * sourceValueNotIn but not both.
-           */
-          if (!ruleNode.has(JSON_TARGETVALUE_TAG)
-              || (!ruleNode.has(JSON_SOURCE_VALUE_IN_TAG)
-                  && !ruleNode.has(JSON_SOURCE_VALUE_NOTIN_TAG))
-              || (ruleNode.has(JSON_SOURCE_VALUE_IN_TAG)
-                  && ruleNode.has(JSON_SOURCE_VALUE_NOTIN_TAG))) {
-            parsingError = true;
-            break;
-          }
-          JsonNode targetValueNode = ruleNode.path(JSON_TARGETVALUE_TAG);
-          String targetValue = targetValueNode.asText();
-
-          // System.out.println("=======> rule: " +
-          // ruleNode.toString());
-          // System.out.println("=======> JSON_TARGETVALUE_TAG: " +
-          // ruleNode.path(JSON_TARGETVALUE_TAG));
-          GeneralizeRule ruleSet = new GeneralizeRule(targetValue);
-
-          // System.out.println("=======> JSON_SOURCE_VALUE_IN_TAG: "
-          // + ruleNode.path(JSON_SOURCE_VALUE_IN_TAG));
-          // System.out.println("=======> JSON_SOURCE_VALUE_NOTIN_TAG:
-          // "
-          // + ruleNode.path(JSON_SOURCE_VALUE_NOTIN_TAG));
-
-          JsonNode sourceValueInList = ruleNode.path(JSON_SOURCE_VALUE_IN_TAG);
-          JsonNode sourceValueNotInList = ruleNode.path(JSON_SOURCE_VALUE_NOTIN_TAG);
-
-          if (ruleNode.has(JSON_SOURCE_VALUE_IN_TAG)) {
-            ruleSet.setLogicalNegation(false);
-            for (JsonNode sourceValueIn : sourceValueInList) {
-              String value = sourceValueIn.asText();
-              ruleSet.getValueSet().add(value);
-              // System.out.println("=======> sourceValueIn: " +
-              // value);
-
-            }
-          } else {
-            // sourceValueNotIn node:
-            ruleSet.setLogicalNegation(true);
-            for (JsonNode sourceValueNotIn : sourceValueNotInList) {
-              String value = sourceValueNotIn.asText();
-              ruleSet.getValueSet().add(value);
-              // System.out.println("=======> sourceValueNotIn: "
-              // + value);
-            }
-          }
-          rulesetList.add(ruleSet);
-        }
-      } else {
-        parsingError = true;
-      }
-      if (parsingError) {
-        throw new RuntimeException();
-      }
+      generalizeRuleSet = configuration.parseMaskRuleSet(configuration.getMaskRuleSet());
     } catch (Exception e) {
-      String message = GENERALIZE_MASK_RULESET + ", ruleSet: " + this.generalizeMaskRuleSet;
+      // should not occur as validation has already checked done a parse

```

Comment:
Typo at the end of the comment?

Reply:
Maybe should be a comma or slash after "checked"